src/test/java/com/alkemy/ong/bigtest/auth/RegisterAuthenticationIntegrationTest.java

```diff
@@ -0,0 +1,235 @@
+                .password("pass1234")
+                .build()))
+            .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(jsonPath("$.id", notNullValue()))
+        .andExpect(jsonPath("$.firstName", equalTo("Pepe")))
+        .andExpect(jsonPath("$.lastName", equalTo("Le Pew")))
+        .andExpect(jsonPath("$.email", equalTo("pepeLePew@mail.com")))
+        .andExpect(jsonPath("$.token", notNullValue()))
+        .andExpect(status().isCreated());
+
+
+      UserEntity newUser = userRepository.findByEmail("pepeLePew@mail.com");
+      assertTrue(newUser != null);
+      assertEquals("Pepe", newUser.getFirstName());
+      assertEquals("Le Pew", newUser.getLastName());
+  }
+
+  @Test
+  public void shouldReturnBadRequestWhenFirstNameIsNull() throws Exception {
+
+    mockMvc.perform(post("/auth/register")
+            .content(objectMapper.writeValueAsString(RegisterRequest.builder()
+                .lastName("Le Pew")
+                .email("pepeLePew@mail.com")
+                .password("pass1234")
+                .build()))
+            .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(jsonPath("$.statusCode", equalTo(400)))
+        .andExpect(jsonPath("$.message", equalTo("Invalid input data.")))
+        .andExpect(jsonPath("$.moreInfo", hasSize(1)))
+        .andExpect(jsonPath("$.moreInfo",
+            hasItem("must not be null")))
+        .andExpect(status().isBadRequest());
+  }
+
+  @Test
+  public void shouldReturnBadRequestWhenFirstNameContainsNumbers() throws Exception {
+
+    mockMvc.perform(post("/auth/register")
+            .content(objectMapper.writeValueAsString(RegisterRequest.builder()
+                .firstName("Pepe 100")
+                .lastName("Le Pew")
+                .email("pepeLePew@mail.com")
+                .password("pass1234")
+                .build()))
+            .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(jsonPath("$.statusCode", equalTo(400)))
+        .andExpect(jsonPath("$.message", equalTo("Invalid input data.")))
+        .andExpect(jsonPath("$.moreInfo", hasSize(1)))
+        .andExpect(jsonPath("$.moreInfo",
+            hasItem("Name can contain letters and spaces")))
+        .andExpect(status().isBadRequest());
+  }
+
+  @Test
+  public void shouldReturnBadRequestWhenLastNameIsNull() throws Exception {
+
+    mockMvc.perform(post("/auth/register")
+            .content(objectMapper.writeValueAsString(RegisterRequest.builder()
+                .firstName("Pepe")
+                .email("pepeLePew@mail.com")
+                .password("pass1234")
+                .build()))
+            .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(jsonPath("$.statusCode", equalTo(400)))
+        .andExpect(jsonPath("$.message", equalTo("Invalid input data.")))
+        .andExpect(jsonPath("$.moreInfo", hasSize(1)))
+        .andExpect(jsonPath("$.moreInfo",
+            hasItem("must not be null")))
```

Comment:
Hay un bug en el DTO porque "must not be null" no es un mensaje de error representativo, debería indicar el nombre del campo. Ejemplo: "firstName must not be null."